src/mlpack/methods/ann/activation_functions/softplus_function.hpp

```diff
@@ -47,13 +47,14 @@ class SoftplusFunction
    * Computes the softplus function.
    *
    * @param x Input data.
+   * @param threshold The threshold value above which softplus behaves linear.
    * @return f(x).
    */
-  static double Fn(const double x)
+  static double Fn(const double x, const double threshold = 40.0)
   {
-    if (x > 20.0)
+    if (x > threshold)
       return x;
-    else 
+    else
       return std::log(1 + std::exp(x));
   }
```

Comment:
Personally, I don't think using threshold as an argument in `Fn` is a great way. What might have been a way to do this is to define a variable in `private` and implement getter and setter methods (with legit default threshold value). And if no threshold value is to be used at all then the implementation could be -

```
static double Fn(const double x)
{
  if (x > 0)
    return std::log(1 + std::exp(-x)) + x;
  else
    return std::log(1 + std::exp(x));
} 
```
You can also use fast approximation of e^-x, referenced in one of my previous comments. Experimenting might reveal if approximation is better choice or not? Note that this is just to evade unnecessary clutter of using `threshold`.